### PR TITLE
SSCS-10349 populate case access management fields on ManualCaseCreatedHandler callback

### PR DESCRIPTION
**Pull request by wmarshid**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10349

### Change description ###

Sets the case access management fields on `CallBack<SscsCaseData>` in the `ManualCaseCreatedHandler` using a new helper class `CaseAccessManagementFieldsHelper`.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import static com.microsoft.applicationinsights.boot.dependencies.apachecommons.lang3.StringUtils.isNotBlank;
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.Benefit.getBenefitOptionalByCode;
+
+import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
+import uk.gov.hmcts.reform.sscs.ccd.domain.FormType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsType;
+
+public class WorkAllocationFieldsHelper {
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java:12]: Not too sure these fields are for work allocation e.g. OGD type is SSCS5. I might be wrong but I think these are just the common components fields.

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -159,6 +159,7 @@ feature:
   secure-doc-store.enabled: ${SECURE_DOC_STORE_FEATURE:false}
   enhancedConfidentiality.enabled: ${ENHANCED_CONFIDENTIALITY_FEATURE:false}
   gaps-switchover.enabled: ${GAPS_SWITCHOVER_FEATURE:false}
+  work-allocation.enabled: ${WORK_ALLOCATION_FEATURE:false}
```
Comment (ss-evoco) [inline src/main/resources/application.yaml:162]: Is this copied over from another PR / branch ?

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ManualCaseCreatedHandler.java [hunk 3]
```diff
@@ -1,61 +1,54 @@
...
-        this.dispatchPriority = DispatchPriority.LATEST;
-        this.ccdService = ccdService;
-        this.idamService = idamService;
-        hmctsServiceIdMap = new HashMap<>();
-        hmctsServiceIdMap.put("HMCTSServiceId", "BBA3");
-        supplementaryDataRequestMap = new HashMap<>();
-        supplementaryDataRequestMap.put("$set", hmctsServiceIdMap);
-    }
+    @Value("${feature.work-allocation.enabled}")
+    private boolean workAllocationFeature;
```
Comment (ss-evoco) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ManualCaseCreatedHandler.java:36]: Is this the feature flag we are trying to use for this (case access management) release ?

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java (hunk 1 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java:12]: The fields we are setting are in the WA fields class, the bug was raised from the WA tab in ExUI (I believe? Not sure.) So I assumed this was related to WA.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (wmarshid) [reply src/main/resources/application.yaml:162]: Nope I added this, can change if needed.

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ManualCaseCreatedHandler.java (hunk 3 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ManualCaseCreatedHandler.java:36]: Potentially yes, can be changed if needed.

Comment (ss-evoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/ManualCaseCreatedHandler.java:36]: Please do. We will have a work allocation feature flag as well and we cannot confuse the two.

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java (hunk 1 above)
Comment (ss-evoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/helper/WorkAllocationFieldsHelper.java:12]: Please confirm with the QA who raised it but I dont believe so. WA isnt a thing yet so I dont think they would have been testing it.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (ss-evoco) [reply src/main/resources/application.yaml:162]: Same as above. Please change to some case access management flag.

**Review by ss-evoco — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yaml (hunk 2 above)
Comment (wmarshid) [reply src/main/resources/application.yaml]: Will be addressed in this PR: https://github.com/hmcts/sscs-common/pull/1178/files

**Review by ss-evoco — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Pull request merged**